build/repo.targets

```diff
@@ -0,0 +1,60 @@
+<Project>
+  <ItemGroup>
+    <Scenarios Include="-n Plaintext -o KestrelHttpServer@dev --kestrelThreadCount 1" />
+    <Scenarios Include="-n Plaintext -o KestrelHttpServer@feature/dev-si --kestrelThreadCount 2" />
+    <Scenarios Include="-n Json -o KestrelHttpServer@dev" />
+    <Scenarios Include="-n Json -o KestrelHttpServer@feature/dev-si" />
+    <Scenarios Include="-n Plaintext" />
+    <Scenarios Include="-n Plaintext -r Desktop" />
+    <Scenarios Include="-n Plaintext --webHost HttpSys" />
+    <Scenarios Include="-n Plaintext -f Benchmarks.PassthroughConnectionFilter" />
+    <Scenarios Include="-n Plaintext -m https" />
+    <Scenarios Include="-n Plaintext -m https -r Desktop" />
+    <Scenarios Include="-n Plaintext -m https --webHost HttpSys" />
+    <Scenarios Include="-n Json" />
+    <Scenarios Include="-n Json -r Desktop" />
+    <Scenarios Include="-n Json --webHost HttpSys" />
+    <Scenarios Include="-n Json -f Benchmarks.PassthroughConnectionFilter" />
+    <Scenarios Include="-n Json -m https" />
+    <Scenarios Include="-n Json -m https -r Desktop" />
+    <Scenarios Include="-n Json -m https --webHost HttpSys" />
+    <Scenarios Include="-n MvcPlaintext" />
+    <Scenarios Include="-n MvcPlaintext -r Desktop" />
+    <Scenarios Include="-n MvcJson" />
+    <Scenarios Include="-n MvcJson -r Desktop" />
+    <Scenarios Include="-n MemoryCachePlaintext" />
+    <Scenarios Include="-n MemoryCachePlaintextSetRemove" />
+    <Scenarios Include="-n ResponseCachingPlaintextCached" />
+    <Scenarios Include="-n ResponseCachingPlaintextCached --method DELETE" />
+    <Scenarios Include="-n ResponseCachingPlaintextResponseNoCache" />
+    <Scenarios Include="-n ResponseCachingPlaintextRequestNoCache" />
+    <Scenarios Include="-n ResponseCachingPlaintextVaryByCached" />
+  </ItemGroup>
+
+  <Target Name="TestBenchmarks" DependsOnTargets="Compile">
+    <Error Text="BENCHMARK_SERVER must be set" Condition="'$(BENCHMARK_SERVER)' == ''" />
+    <Error Text="BENCHMARK_CLIENT must be set" Condition="'$(BENCHMARK_CLIENT)' == ''" />
+
+    <GetDotNetHost>
+      <Output TaskParameter="ExecutablePath" PropertyName="DotNetPath" />
+    </GetDotNetHost>
+    <MSBuild Projects="$(MSBuildProjectFullPath)"
+      Targets="RunBenchmark"
+      Properties="Scenario=%(Scenarios.Identity);DotNetPath=$(DotNetPath)" />
+  </Target>
+
+  <Target Name="RunBenchmark">
+    <PropertyGroup>
+      <DotNetPath Condition="'$(DotNetPath)' == ''">dotnet</DotNetPath>
+      <RunArgs>$(DotNetPath) run -- $(Scenario) </RunArgs>
```

Comment:
@natemcmaster: It's not related to your change, but it might be slightly faster to change this from `dotnet run` to `dotnet $(OutputPath)/BenchmarksDriver.dll`.  Is this feasible?  Is there an existing variable that would give us the output path from the previous `Compile` task?

Also, it might be more correct for `RunBenchmark` to depend on `Compile` (rather than `TestBenchmarks`).  Since if someone did directly call `RunBenchmark` once, it should ensure `BenchmarksDriver` has been compiled.

I'd suggest merging what you have now if it works, and we can make these improvements in a separate PR.